### ISSUE TITLE
Error compile on Linux Ubuntu 16.04

### ISSUE THREAD

**Issue by legluondunet** [MoochMcGee/PCem-mooch#10]
Hello,

when I compile it on Linux I obtain this error:

$ make
Making all in src
make[1] : on entre dans le répertoire « /home/legluondunet/Bureau/PCemV10.1Linux/src »
source='386.c' object='pcem-386.o' libtool=no \
DEPDIR=.deps depmode=none /bin/bash ../depcomp \
gcc -DPACKAGE_NAME=\"PCem\" -DPACKAGE_TARNAME=\"pcem\" -DPACKAGE_VERSION=\"v10.1\" -DPACKAGE_STRING=\"PCem\ v10.1\" -DPACKAGE_BUGREPORT=\"Tom\ Walker\ <tommowalker@tommowalker.co.uk>\" -DPACKAGE_URL=\"\" -DPACKAGE=\"pcem\" -DVERSION=\"v10.1\" -DHAVE_LIBOPENAL=1 -DHAVE_LIBALUT=1 -DHAVE_LIBPTHREAD=1 -I.    -I/usr/include -O3 -c -o pcem-386.o `test -f '386.c' || echo './'`386.c
/bin/bash: ../depcomp: Aucun fichier ou dossier de ce type
Makefile:582 : la recette pour la cible « pcem-386.o » a échouée
make[1]: **\* [pcem-386.o] Erreur 127
make[1] : on quitte le répertoire « /home/legluondunet/Bureau/PCemV10.1Linux/src »
Makefile:247 : la recette pour la cible « all-recursive » a échouée
make: **\* [all-recursive] Erreur 1

Thank you for you help.

Config: Ubuntu Xenial 16.04

**Comment by bachimusprime** [comment]
Go into the source directory and use make -f Makefile.linux64

**Comment by legluondunet** [comment]
ok, it begins to compile but I obtain another error:

allegro-main.c:164:2: warning: implicit declaration of function ‘midi_close’ [-Wimplicit-function-declaration]
  midi_close();
  ^
gcc -O3 -march=core2 -mtune=core2 -fomit-frame-pointer -ffast-math -msse -msse2 -msse3 -mssse3 -mfpmath=sse -c allegro-midi.c
gcc -O3 -march=core2 -mtune=core2 -fomit-frame-pointer -ffast-math -msse -msse2 -msse3 -mssse3 -mfpmath=sse -c allegro-mouse.c
gcc -O3 -march=core2 -mtune=core2 -fomit-frame-pointer -ffast-math -msse -msse2 -msse3 -mssse3 -mfpmath=sse -c allegro-video.c
allegro-video.c: In function ‘allegro_video_update_size’:
allegro-video.c:58:17: warning: implicit declaration of function ‘fatal’ [-Wimplicit-function-declaration]
                 fatal("Failed to set gfx mode %i,%i : %s\n", x, y, allegro_erro
                 ^
gcc -O3 -march=core2 -mtune=core2 -fomit-frame-pointer -ffast-math -msse -msse2 -msse3 -mssse3 -mfpmath=sse -c amstrad.c
amstrad.c: In function ‘amstrad_read’:
amstrad.c:17:28: warning: implicit declaration of function ‘readdacfifo’ [-Wimplicit-function-declaration]
                 return 7 | readdacfifo();
                            ^
gcc -O3 -march=core2 -mtune=core2 -fomit-frame-pointer -ffast-math -msse -msse2 -msse3 -mssse3 -mfpmath=sse -c cdrom-ioctl-linux.c
cdrom-ioctl-linux.c: In function ‘ioctl_audio_callback’:
cdrom-ioctl-linux.c:86:2: warning: implicit declaration of function ‘close’ [-Wimplicit-function-declaration]
  close(fd);
  ^
cdrom-ioctl-linux.c: At top level:
cdrom-ioctl-linux.c:193:29: error: unknown type name ‘cdrom_tocentry’
 static int read_toc(int fd, cdrom_tocentry btoc)
                             ^
cdrom-ioctl-linux.c: In function ‘ioctl_ready’:
cdrom-ioctl-linux.c:261:14: warning: implicit declaration of function ‘read_toc’ [-Wimplicit-function-declaration]
   tocvalid = read_toc(fd, toc);
              ^
cdrom-ioctl-linux.c: In function ‘ioctl_get_last_block’:
cdrom-ioctl-linux.c:277:3: error: unknown type name ‘cdrom_tocentry’
   cdrom_tocentry lbtoc[100];
   ^
cdrom-ioctl-linux.c: In function ‘ioctl_readsector’:
cdrom-ioctl-linux.c:426:9: warning: implicit declaration of function ‘lseek’ [-Wimplicit-function-declaration]
         lseek(cdrom, sector_2048, SEEK_SET);
         ^
cdrom-ioctl-linux.c:427:9: warning: implicit declaration of function ‘read’ [-Wimplicit-function-declaration]
         read(cdrom, b, 2048);
         ^
cdrom-ioctl-linux.c: In function ‘ioctl_readsector_raw’:
cdrom-ioctl-linux.c:459:2: error: ‘err’ undeclared (first use in this function)
  err = ioctl(cdrom, CDROMREADRAW, (void *) &raw_read_params);
  ^
cdrom-ioctl-linux.c:459:2: note: each undeclared identifier is reported only once for each function it appears in
cdrom-ioctl-linux.c:463:10: warning: ‘return’ with a value, in function returning void
   return 0;
          ^
Makefile.linux64:193 : la recette pour la cible « cdrom-ioctl-linux.o » a échouée
make: *_\* [cdrom-ioctl-linux.o] Erreur 1

**Comment by bachimusprime** [comment]
it compiles fine for me.

**Comment by bachimusprime** [comment]
Ask @Twinaphex

**Comment by legluondunet** [comment]
you compiled it on Ubuntu 16.04?

**Comment by bachimusprime** [comment]
No, I compiled it on Arch Linux.

**Comment by bachimusprime** [comment]
@legluondunet Try now.

**Comment by legluondunet** [comment]
the compilation did not even begin:

PCem-mooch-master/src$ make -f Makefile.linux64
Makefile.linux64:111: **\* La recette commence avant la première cible. Arrêt.

**Comment by bachimusprime** [comment]
I can't figure out that error because I don't speak French. Also it works
for @twinaphex

**Comment by legluondunet** [comment]
In fact it missed a "\" in the file "Makefile.linux64, line 110. I added it then I can launch the compilation.
But finished with this error:

gcc-5.real: error: resid-fp/convolve-sse.o: Aucun fichier ou dossier de ce type (=no file or folder, can not found it)
Makefile.linux64:185 : la recette pour la cible « PCem64 » a échouée (=PCem64 could not be compiled)
make: **\* [PCem64] Erreur 1
Do you know why "resid-fp/convolve-sse.o" was not compile?

Another thing: I see some references to Automake 1.11.
But Ubuntu 16.06 Xenial have Automake 1.15.

Good news: Libretro module compiled with success! But I didn't test it yet.

**Comment by bachimusprime** [comment]
I told you not to use that. Use Makefile.linux64

**Comment by legluondunet** [comment]
I used it and it gave me the error above. (make -f Makefile.linux64 in src folder).

**Comment by bachimusprime** [comment]
well, it works for @twinaphex bud

**Comment by legluondunet** [comment]
Well I just tried to compile it on another computer with Ubuntu 16.04 Xenial:

src$ make -f Makefile.linux64
...
gcc 386.o 386_dynarec.o 386_dynarec_ops.o 808x.o acer386sx.o acerm3a.o ali1429.o allegro-gui-configure.o allegro-gui-hdconf.o allegro-gui.o allegro-joystick.o allegro-keyboard.o allegro-main.o allegro-midi.o allegro-mouse.o allegro-video.o amstrad.o cdrom-ioctl-linux.o cdrom-iso.o cdrom-null.o codegen.o codegen_ops.o codegen_timing_486.o codegen_timing_pentium.o codegen_x86-64.o compaq.o config.o cpu.o dac.o device.o disc.o disc_fdi.o disc_img.o disc_sector.o dma.o fdc.o fdc37c665.o fdc37c932fr.o fdd.o fdi2raw.o gameport.o headland.o i430hx.o i430lx.o i430fx.o i430nx.o i430vx.o ide.o intel.o intel_flash.o io.o jim.o keyboard.o keyboard_amstrad.o keyboard_at.o keyboard_olim24.o keyboard_pcjr.o keyboard_xt.o linux-time.o lpt.o mcr.o mem.o memregs.o model.o mouse.o mouse_ps2.o mouse_serial.o neat.o nmi.o nvr.o olivetti_m24.o opti.o pc.o pc87306.o pci.o pic.o piix.o pit.o ppi.o ps1.o rom.o serial.o sis496.o sis85c471.o sio.o sound.o sound_ad1848.o sound_adlib.o sound_adlibgold.o sound_cms.o sound_dbopl.o sound_emu8k.o sound_gus.o sound_mpu401_uart.o sound_opl.o sound_pas16.o sound_pssj.o sound_resid.o sound_sb.o sound_sb_dsp.o sound_sn76489.o sound_speaker.o sound_ssi2001.o sound_wss.o sound_ym7128.o soundopenal.o tandy_eeprom.o tandy_rom.o thread-pthread.o timer.o um8669f.o vid_ati_eeprom.o vid_ati_mach64.o vid_ati18800.o vid_ati28800.o vid_ati68860_ramdac.o vid_cga.o vid_cga_comp.o vid_cl5429.o vid_ega.o vid_et4000.o vid_et4000w32.o vid_hercules.o vid_icd2061.o vid_ics2595.o vid_incolor.o vid_mda.o vid_nv_riva128.o vid_nv_rivatnt.o vid_olivetti_m24.o vid_oti067.o vid_paradise.o vid_pc1512.o vid_pc1640.o vid_pc200.o vid_pcjr.o vid_s3.o vid_s3_virge.o vid_sdac_ramdac.o vid_stg_ramdac.o vid_svga.o vid_svga_render.o vid_tandy.o vid_tandysl.o vid_tgui9440.o vid_tkd8001_ramdac.o vid_tvga.o vid_unk_ramdac.o vid_vga.o vid_voodoo.o video.o w83877f.o wd76c10.o x86seg.o x87.o xtide.o dosbox/dbopl.o resid-fp/convolve.o resid-fp/convolve-sse.o resid-fp/envelope.o resid-fp/extfilt.o resid-fp/filter.o resid-fp/pot.o resid-fp/sid.o resid-fp/voice.o resid-fp/wave6581__ST.o resid-fp/wave6581_P_T.o resid-fp/wave6581_PS_.o resid-fp/wave6581_PST.o resid-fp/wave8580__ST.o resid-fp/wave8580_P_T.o resid-fp/wave8580_PS_.o resid-fp/wave8580_PST.o resid-fp/wave.o  -o "PCem64" -lopenal -lalut -flto -lgcov -lstdc++ -lm -lalleg -pthread
gcc: error: dosbox/dbopl.o: Aucun fichier ou dossier de ce type
gcc: error: resid-fp/convolve.o: Aucun fichier ou dossier de ce type
gcc: error: resid-fp/convolve-sse.o: Aucun fichier ou dossier de ce type
gcc: error: resid-fp/envelope.o: Aucun fichier ou dossier de ce type
gcc: error: resid-fp/extfilt.o: Aucun fichier ou dossier de ce type
gcc: error: resid-fp/filter.o: Aucun fichier ou dossier de ce type
gcc: error: resid-fp/pot.o: Aucun fichier ou dossier de ce type
gcc: error: resid-fp/sid.o: Aucun fichier ou dossier de ce type
gcc: error: resid-fp/voice.o: Aucun fichier ou dossier de ce type
gcc: error: resid-fp/wave6581__ST.o: Aucun fichier ou dossier de ce type
gcc: error: resid-fp/wave6581_P_T.o: Aucun fichier ou dossier de ce type
gcc: error: resid-fp/wave6581_PS_.o: Aucun fichier ou dossier de ce type
gcc: error: resid-fp/wave6581_PST.o: Aucun fichier ou dossier de ce type
gcc: error: resid-fp/wave8580__ST.o: Aucun fichier ou dossier de ce type
gcc: error: resid-fp/wave8580_P_T.o: Aucun fichier ou dossier de ce type
gcc: error: resid-fp/wave8580_PS_.o: Aucun fichier ou dossier de ce type
gcc: error: resid-fp/wave8580_PST.o: Aucun fichier ou dossier de ce type
gcc: error: resid-fp/wave.o: Aucun fichier ou dossier de ce type
Makefile.linux64:185 : la recette pour la cible « PCem64 » a échouée
make: **\* [PCem64] Erreur 1

and there is stil the error in Makefile.linux64 line 110, I have to manually corrected it before to compile.

PCEM compiles ok until it needs components from dosbox and resid-fp

Thank you for your help.

**Comment by legluondunet** [comment]
I finally found the missed files dbopl.o convolve-sse.o convolve.o in the main src folder while Makefile searched them in the dosbox and resid-fp folders. I move them and now I have this error message:

gcc: error: resid-fp/envelope.o: Aucun fichier ou dossier de ce type
gcc: error: resid-fp/extfilt.o: Aucun fichier ou dossier de ce type
gcc: error: resid-fp/filter.o: Aucun fichier ou dossier de ce type
gcc: error: resid-fp/pot.o: Aucun fichier ou dossier de ce type
gcc: error: resid-fp/sid.o: Aucun fichier ou dossier de ce type
gcc: error: resid-fp/voice.o: Aucun fichier ou dossier de ce type
gcc: error: resid-fp/wave6581__ST.o: Aucun fichier ou dossier de ce type
gcc: error: resid-fp/wave6581_P_T.o: Aucun fichier ou dossier de ce type
gcc: error: resid-fp/wave6581_PS_.o: Aucun fichier ou dossier de ce type
gcc: error: resid-fp/wave6581_PST.o: Aucun fichier ou dossier de ce type
gcc: error: resid-fp/wave8580__ST.o: Aucun fichier ou dossier de ce type
gcc: error: resid-fp/wave8580_P_T.o: Aucun fichier ou dossier de ce type
gcc: error: resid-fp/wave8580_PS_.o: Aucun fichier ou dossier de ce type
gcc: error: resid-fp/wave8580_PST.o: Aucun fichier ou dossier de ce type
gcc: error: resid-fp/wave.o: Aucun fichier ou dossier de ce type
Makefile.linux64:185 : la recette pour la cible « PCem64 » a échouée
make: **\* [PCem64] Erreur 1

Same soluce: this files are located in the main pcem src folder, I moved in their respective directories.

relaunch compilation, again the same error type:
.o resid-fp/wave8580__ST.o resid-fp/wave8580_P_T.o resid-fp/wave8580_PS_.o resid-fp/wave8580_PST.o resid-fp/wave.o  -o "PCem64" -lopenal -lalut -flto -lgcov -lstdc++ -lm -lalleg -pthread
gcc: error: resid-fp/wave6581_PS_.o: Aucun fichier ou dossier de ce type
gcc: error: resid-fp/wave6581_PST.o: Aucun fichier ou dossier de ce type
gcc: error: resid-fp/wave8580__ST.o: Aucun fichier ou dossier de ce type
gcc: error: resid-fp/wave8580_P_T.o: Aucun fichier ou dossier de ce type
gcc: error: resid-fp/wave8580_PS_.o: Aucun fichier ou dossier de ce type
gcc: error: resid-fp/wave8580_PST.o: Aucun fichier ou dossier de ce type
gcc: error: resid-fp/wave.o: Aucun fichier ou dossier de ce type

same soluce applied...and PCEM finally compiled!
